operators/src/upgrade/rest/service.rs

```diff
@@ -125,6 +138,234 @@ pub async fn apply_upgrade() -> Result<HttpResponse, RestError> {
     }
 }
 
+/// Upgrade data plane by controlled restart of io-engine pods
+pub async fn upgrade_data_plane() -> Result<(), Error> {
+    let pods: Api<Pod> = Api::namespaced(
+        UpgradeConfig::get_config().k8s_client().client(),
+        UpgradeConfig::get_config().namespace(),
+    );
+
+    let io_engine_listparam = ListParams::default().labels(IO_ENGINE_POD_LABEL);
+    let initial_io_engine_pod_list: ObjectList<Pod> = pods.list(&io_engine_listparam).await?;
+    for pod in initial_io_engine_pod_list.iter() {
+        // Fetch the node name on which the io-engine pod is running
+        let node_name = pod
+            .spec
+            .as_ref()
+            .ok_or_else(|| anyhow::anyhow!("pod.spec is empty"))?
+            .node_name
+            .as_ref()
+            .ok_or_else(|| anyhow::anyhow!("pod.spec.node_name is empty"))?
+            .as_str();
+
+        tracing::info!(
+            pod.name = %pod.metadata.name.clone().unwrap_or_default(),
+            node.name = %node_name,
+            "Upgrade starting for data-plane pod"
+        );
+
+        let is_node_cordoned = is_node_cordoned(node_name).await?;
+
+        // Issue node drain command
+        issue_node_drain(node_name).await?;
+
+        // Wait for node drain to complete across the cluster.
+        wait_node_drain().await?;
+
+        // Wait for any rebuild to complete.
+        wait_for_rebuild().await?;
+
+        // restart the data plane pod
+        restart_data_plane(&node_name, pod).await?;
+
+        // Uncordon the drained node
+        if !is_node_cordoned {
+            uncordon_node(node_name).await?;
+        }
+
+        // validate the new pod is up and running
+        verify_data_plane_pod_is_running(node_name).await?;
+
+        // Validate the control plane pod is up and running
+        is_control_plane_running().await?
+    }
+    Ok(())
+}
+
+pub async fn uncordon_node(node_name: &str) -> Result<(), Error> {
+    match UpgradeConfig::get_config()
+        .rest_client()
+        .nodes_api()
+        .delete_node_cordon(node_name, DRAIN_FOR_UPGRADE)
+        .await
+    {
+        Ok(_) => {
+            tracing::info!(node.name = node_name, "Node is uncordoned");
+        }
+        Err(_) => {
+            Error::NodeUncordonError {
+                node_name: node_name.to_string(),
+            };
+        }
+    }
+    Ok(())
```

Comment:
Semicolon here is not right:

```suggestion
    {
        Ok(_) => {
            tracing::info!(node.name = node_name, "Node is uncordoned");
            Ok(())
        }
        Err(_) => {
            Error::NodeUncordonError {
                node_name: node_name.to_string(),
            }
        }
    }
```